### ISSUE TITLE
Add support for page-break-before|after: left|right

### ISSUE THREAD

**Issue by PhenX** [dompdf/dompdf#546]
_Original author: rschu (September 27, 2012 17:17:30)_

For duplex printing, having support for left/right page-break rules would be helpful. When printing many individual items as a single document, using duplex printing mode, simply breaking to the next page is not enough, because the next item might appear on the back-side of the previous item's last page. The break left/right rules enable you to break to the next odd or even page - e.g. inserting a blank right page if you're breaking from a left page to another left page, and vice-versa.

Note that no browser currently seems to properly support this rule, although it has been around for a long while - browser vendors do not seem to be very concerned about printing, but presumably that's the shoe that dompdf attempts to fill? :-)

_Original issue: http://code.google.com/p/dompdf/issues/detail?id=540_

**Comment by bramceulemans** [comment]
Any update on this?

**Comment by bsweeney** [comment]
Nothing yet. Reposing from 2337 ...

Dompdf currently treats left/right as "always" when doing page-break calculations. While this is valid per the CSS 2 spec ([ref](https://www.w3.org/TR/CSS2/page.html#propdef-page-break-before)), which states:

> A conforming user agent may interpret the values 'left' and 'right' as 'always'.

it would be more useful to accommodate the definition of left/right as stated:

- left: Force one or two page breaks before (after) the generated box so that the next page is formatted as a left page.
- right: Force one or two page breaks before (after) the generated box so that the next page is formatted as a right page.

**Comment by bramceulemans** [comment]
I agree, it would be more useful. Especially when I want to render a page for double sided printing.
